src/app/api/auth/lib/nextauth-options.ts

```diff
@@ -22,6 +32,69 @@ export const verifySession = async (token: string | undefined) => {
   }
 }
 
+/**
+ * Refresh access token
+ * @param clientId Client ID
+ * @param clientSecret Client Secret
+ * @param refreshToken Refresh token
+ * @returns object — New access token and refresh token
+ */
+export const refreshAccessToken = async (
+  token: JWT,
+  clientId: string,
+  clientSecret: string,
+  refreshToken: string,
+) => {
+  try {
+    authLogger.debug('Refreshing access token', clientId)
+
+    const url =
+      'https://github.com/login/oauth/access_token?' +
+      new URLSearchParams({
+        client_id: clientId,
+        client_secret: clientSecret,
+        refresh_token: refreshToken,
+        grant_type: 'refresh_token',
+      })
+
+    const response = await fetch(url, {
+      headers: {
+        'Content-Type': 'application/x-www-form-urlencoded',
+      },
+      method: 'POST',
+    }).then(async (res) => {
+      const responseText = await res.text()
+      const entries = new URLSearchParams(responseText)
+      const params = Object.fromEntries(entries)
+
+      if (params.error) {
+        throw new Error(params.error_description)
+      }
+
+      return params
+    })
+
+    return {
+      accessToken: response.access_token,
+      // Access token expiration is provided as number in seconds until expiration (value is always 8 hours)
+      accessTokenExpires:
+        Date.now() + secondsToMilliseconds(Number(response.expires_in)),
```

Comment:
I'm thinking we should convert the `secondsToMilliseconds` func to include appending the `Date.now()` since that will shore up some of this code. Like `normalizeExpireDates`

(and the function name itself is longer than just `* 1000` 😄)

Reply:
I could add it, but there are a couple of the dates that don't get Date.now() added because they get transformed by `next-auth`.